### PR TITLE
feat: Introduce Customizable Pod Admission FailurePolicy

### PR DESCRIPTION
**Pull request by dacamposol**
Relates to [opentelemetry-operator#1765](https://github.com/open-telemetry/opentelemetry-operator/issues/1765)

### CODE REVIEW

**Review by TylerHelmuth — COMMENTED** [review]
@dacamposol thanks for working through that issue.  Please run `make generate-examples CHARTS=opentelemetry-operator` to generate the examples.

File: charts/opentelemetry-operator/values.yaml [hunk 1]
```diff
@@ -186,6 +186,12 @@ admissionWebhooks:
   failurePolicy: Fail
   secretName: ""
 
+  ## Defines the sidecar injection logic in Pods.
+  ## - Ignore, the injection is fail-open. The pod will be created, but the sidecar won't be injected.
+  ## - Fail, the injection is fail-close. If the webhook pod is not ready, pods cannot be created.
+  pods:
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/values.yaml:192]: @dacamposol please update the json schema file.

Comment (dacamposol) [reply charts/opentelemetry-operator/values.yaml:192]: Done, but just for information, the JSON Schema isn't following your own defined configuration.

From what I see, the indentation is of 4 spaces in the existing file, but afterwards in `.editorconfig` you have defined the indent to be 2:

```.editorconfig
[*.{json,yaml}]
indent_style = space
indent_size = 2
```

Are you maybe autogenerating that file and not formatting it afterwards?

Comment (TylerHelmuth) [reply charts/opentelemetry-operator/values.yaml:192]: Yes I think it was being auto-generated for a time.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**